crates/mcp-client/src/service.rs

```diff
@@ -50,24 +50,26 @@ where
         let pending_requests = self.pending_requests.clone();
 
         Box::pin(async move {
-            match request {
-                JsonRpcMessage::Request(JsonRpcRequest { id: Some(id), .. }) => {
+            match &request {
+                JsonRpcMessage::Request(JsonRpcRequest { id, .. }) => {
                     // Create a channel to receive the response
                     let (sender, receiver) = oneshot::channel();
                     pending_requests.insert(id.to_string(), sender).await;
 
                     transport.send(request).await?;
                     receiver.await.map_err(|_| Error::ChannelClosed)?
                 }
-                JsonRpcMessage::Request(_) => {
-                    // Handle notifications without waiting for a response
-                    transport.send(request).await?;
-                    Ok(JsonRpcMessage::Nil)
-                }
                 JsonRpcMessage::Notification(_) => {
                     // Handle notifications without waiting for a response
                     transport.send(request).await?;
-                    Ok(JsonRpcMessage::Nil)
+                    // Return a dummy response for notifications
+                    let dummy_response: JsonRpcMessage =
+                        JsonRpcMessage::Response(rmcp::model::JsonRpcResponse {
+                            jsonrpc: rmcp::model::JsonRpcVersion2_0,
+                            id: rmcp::model::RequestId::Number(0),
+                            result: serde_json::Map::new(),
+                        });
+                    Ok(dummy_response)
```

Comment:
I was going to suggest we make the message optional now that Nil isn't a valid JsonRpcMessage, but this code will all go away soon anyway, so maybe not that important...